src/numeric_index.c

```diff
@@ -79,20 +79,21 @@ static size_t NumericRange_Add(NumericRange *n, t_docId docId, double value) {
  * but if we see performance issues in the future, we can consider using another algorithm
  * like QuickSelect or an approximation algorithm for the median.
  */
-static double NumericRange_GetMedian(QueryIterator *iter, size_t num_entries) {
+static double NumericRange_GetMedian(IndexReader *reader, size_t num_entries) {
   size_t median_idx = num_entries / 2;
```

Comment:
Is it possible to get the number of entries from the reader? It wasn't possible with a generic iterator, but a reader always has an index that it reads. I think it makes the API clearer
```suggestion
static double NumericRange_GetMedian(IndexReader *reader) {
  size_t median_idx = InvertedIndex_NumEntries(reader->idx) / 2;
```

Reply:
Indeed, that's easier.